plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/ServicesAndRoutingPage.java

```diff
@@ -213,6 +203,37 @@ public int compare(IServicePort first, IServicePort second) {
 			
 		};
 	}
+
+	private static final Random seed = new Random();
+
+	private static int generateNewPort(Set<Integer> ports) {
+		while(true) {
+			int n = 1025 + seed.nextInt(65536 - 1025); // port should be in range [1025..65535]
+			if(!ports.contains(n)) {
+				return n;
+			}
+		}
+	}
+
+	/**
+	 * Returns integer if string value is an integer in range [0..65535], or -1.
+	 * @param stringPort
+	 * @return
+	 */
+	private static int getIntegerPort(String stringPort) {
+		if(StringUtils.isBlank(stringPort) || stringPort.length() > 5) {
+			return -1;
+		}
+		int value = 0;
+		for (int i = 0; i < stringPort.length(); i++) {
```

Comment:
Wouldnt it be easier to utilize StringUtils#isNumeric in conjuctions With Integer.parse ?